### ISSUE TITLE
Error retrieving match on .Net Core 3.0

### ISSUE THREAD

**Issue by Djipyy** [MingweiSamuel/Camille#23]
Using the following code, I get an exception only on .Net Core 3.0. 
The code works fine on .Net Core 2.1

It happens even with simple code like that, but it also happens on the exemple from the README : 

> using MingweiSamuel.Camille;
using System;
namespace testtt
{
    class Program
    {
        public static string riotApiKey = "--";
       static void Main(string[] args)
        {
            var riotApi = RiotApi.NewInstance(riotApiKey);
            var testmatch = riotApi.MatchV4.GetMatch(MingweiSamuel.Camille.Enums.Region.EUW, (long)4257164534);
        }
    }
}

This is the exception : 

> System.AggregateException
  HResult=0x80131500
  Message=One or more errors occurred. (Unable to cast object of type 'MingweiSamuel.Camille.MatchV4.Participant' to type 'MingweiSamuel.Camille.MatchV4.Match'.)
  Source=System.Private.CoreLib
  Arborescence des appels de procédure :
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at MingweiSamuel.Camille.RiotApi.Get[T](String methodId, String url, Region region, IEnumerable`1 queryParams, Boolean nonRateLimited, Nullable`1 token)
   at MingweiSamuel.Camille.MatchV4Endpoints.GetMatch(Region region, Int64 matchId, Nullable`1 cancellationToken)
   at testtt.Program.Main(String[] args) in C:\Users\djipyy\source\repos\testtt\Program.cs:line 13
 Exception interne 1 :
InvalidCastException : Unable to cast object of type 'MingweiSamuel.Camille.MatchV4.Participant' to type 'MingweiSamuel.Camille.MatchV4.Match'.

**Comment by RyadaProductions** [comment]
Be aware that if this was your API key that you should regenerate it asap.
Other than that I will take a look at the problem

Message information:
 Path: $.gameMode | LineNumber: 0 | BytePositionInLine: 178.)

InnerStacktrace:
```
   at System.Text.Json.ThrowHelper.ThrowJsonException()
   at System.Text.Json.Serialization.Converters.JsonConverterEnum`1.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)
   at System.Text.Json.JsonPropertyInfoNotNullable`4.OnRead(JsonTokenType tokenType, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonPropertyInfo.Read(JsonTokenType tokenType, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.HandleValue(JsonTokenType tokenType, JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
   at System.Text.Json.JsonSerializer.ReadCore(Type returnType, JsonSerializerOptions options, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.ParseCore(String json, Type returnType, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
   at MingweiSamuel.Camille.Util.RegionalRequester.<Get>d__8`1.MoveNext() in C:\Projects\Camille\Camille\src\Util\RegionalRequester.cs:line 94
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at MingweiSamuel.Camille.Util.RequestManager.<Get>d__4`1.MoveNext() in C:\Projects\Camille\Camille\src\Util\RequestManager.cs:line 37
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at MingweiSamuel.Camille.RiotApi.<GetAsync>d__20`1.MoveNext() in C:\Projects\Camille\Camille\src\RiotApi.cs:line 55
```

**Comment by RyadaProductions** [comment]
It seems like that inside of CoreCLR `System.Text.Json` tries to generate a wrong type and then when it returns its trying to cast it to the actual requested type. I will be trying a few things throughout the weekend to see if i can resolve this or find a definite cause. For now i would recommend staying on net core 2.1 since there we still use Newtonsoft.Json

**Comment by RyadaProductions** [comment]
This issue is not present in 3.1 preview 2.
3.0 and 3.1 preview 1 still contain the faulty behavior. If you want to use 3.0 i would recommend using the 3.1 preview 2 or waiting for the 3.1 release.

**Comment by MingweiSamuel** [comment]
Should we bump up `USE_SYSTEXTJSON` to core 3.1+ or do you think 3.0 will be patched?

**Comment by RyadaProductions** [comment]
I think Bumping it to 3.1 is a safe bet for now, we just have to remember that 3.0 still needs to be using Newtonsoft.Json untill it is either patched or .net 5.0 hits.

**Comment by RyadaProductions** [comment]
Created a PR for the version bump. This should effectively solve the issue with the wrong type being returned. Which has been confirmed to be a .net core bug outside of our control.

**Comment by MingweiSamuel** [comment]
I released 2.4.1 with the 3.1 bump, untested.

**Comment by MingweiSamuel** [comment]
Hopefully that version should work @Djipyy 

Once the underlying issue is figured out we can figure out if/where `[System.Text.Json.Serialization.JsonConverter(typeof(System.Text.Json.Serialization.JsonStringEnumConverter))]` is needed for `system.text.json`

**Comment by MingweiSamuel** [comment]
Fixed by #25 v3 e5f7aad445097d6caf04e44691cce986cdaea83d v2 1e25ae778115223913683f791782120f9bfd6649 some time ago

Additionally .NET Core 3.0 is EOL, replaced by 3.1
